dcicutils/ff_utils.py

```diff
@@ -457,65 +457,150 @@ def faceted_search(key=None, ff_env=None, item_type=None, **kwargs):
+                            qc_metrics.update(qc_info)
+
+                    else:
+                        qc_uuid = entry['quality_metric']['uuid']
+                        qc_info = {qc_uuid: {}}
+                        qc_meta = get_metadata(qc_uuid, key=key, ff_env=ff_env)
+                        qc_info[qc_uuid]['values'] = {k: v for k, v in qc_meta.items() if k not in ignored_qc_fields}
+                        qc_info[qc_uuid]['source_file_association'] = associated_file if associated_file != 'files' else 'raw_file'
+                        qc_info[qc_uuid]['source_file'] = entry['accession']
+                        qc_info[qc_uuid]['source_file_type'] = entry['file_type_detailed']
+                        qc_metrics.update(qc_info)
+    return qc_metrics
+
+
+def get_associated_qc_metrics(uuid, key=None, ff_env=None, include_processed_files=True,
+                              include_raw_files=False,
+                              include_supplementary_files=False):
+    """
+    Given a uuid of an experimentSet return a dictionary of dictionaries with each dictionary
+    representing a quality metric.
 
     Args:
-        exclude_raw_files: if False will provide QC metrics on raw files as well
-                           Default: True
-        exclude_supplementary_files: if False will also give QC's associated with
-                                     non-processed files. Default: True
+        include_processed_files: if False will exclude QC metrics on processed files
+                                Default: True
+        include_raw_files: if True will provide QC metrics on raw files as well
+                           Default: False
+        include_supplementary_files: if True will also give QC's associated with
+                                     non-processed files. Default: False
+    Returns:
+        a dictionary of dictionaries with the following structure:
+            {'qc_metric_uuid'}:{'values':'', The values of the qc_metric object
+                                'source_file_association': the file class (processed_file or raw_files)
+                                'source_file':, the accession of the file that the qc is linked to
+                                'source_file_type', the description of the file that the qc is linked to
+                                'experiment_description', the description of the experiment or experimentset
+                                'organism', the organism
+                                'experiment_type', the experiment type (in situ Hi-C, ChIP-seq)
+                                'experiment_subclass' (Hi-C)
+                                'source_experiment': the experiment the qc is linked to (if apply)
+                                'source_experimentSet': the experimentSet the qc is linked to
+                                'biosource_summary': the experiment biosource
```

Comment:
Something seems garbled here. Braces are unbalanced and in the wrong place. There's an extra empty string. Some commas in wrong places. I think you mean to say 
```
            {
                <qc_metric_uuid>: {
                    'values': <the values of the qc_metric object>,
                    'source_file_association': <the file class (processed_file or raw_files)>,
                    'source_file': <the accession of the file that the qc is linked to>,
                   ...etc.
                }
            }
```
I didn't fill them all out. You'll want to replace the "...etc." part. But I think the "<...>" notation is common for meta-syntactic identifiers that describe what goes in a position.